runtime/src/epoch_stakes.rs

```diff
@@ -1,15 +1,81 @@
 use {
     crate::stakes::SerdeStakesToStakeFormat,
     serde::{Deserialize, Serialize},
+    solana_bls_signatures::{Pubkey as BLSPubkey, PubkeyCompressed as BLSPubkeyCompressed},
     solana_clock::Epoch,
     solana_pubkey::Pubkey,
     solana_vote::vote_account::VoteAccountsHashMap,
-    std::{collections::HashMap, sync::Arc},
+    std::{
+        collections::HashMap,
+        sync::{Arc, OnceLock},
+    },
 };
 
 pub type NodeIdToVoteAccounts = HashMap<Pubkey, NodeVoteAccounts>;
 pub type EpochAuthorizedVoters = HashMap<Pubkey, Pubkey>;
 
+#[derive(Clone, Debug, Default)]
+#[cfg_attr(feature = "frozen-abi", derive(AbiExample))]
+#[cfg_attr(feature = "dev-context-only-utils", derive(PartialEq))]
+pub struct BLSPubkeyToRankMap {
+    rank_map: HashMap<BLSPubkey, u16>,
+    //TODO(wen): We can make SortedPubkeys a Vec<BLSPubkey> after we remove ed25519
+    // pubkey from certificate pool.
+    sorted_pubkeys: Vec<(Pubkey, BLSPubkey)>,
+}
+
+impl BLSPubkeyToRankMap {
+    pub fn new(epoch_vote_accounts_hash_map: &VoteAccountsHashMap) -> Self {
+        let mut pubkey_stake_pair_vec: Vec<(Pubkey, BLSPubkey, u64)> = epoch_vote_accounts_hash_map
+            .iter()
+            .filter_map(|(pubkey, (stake, account))| {
+                if *stake > 0 {
+                    account
+                        .vote_state_view()
+                        .bls_pubkey_compressed()
+                        .and_then(|bls_pubkey_compressed_bytes| {
+                            let bls_pubkey_compressed =
+                                BLSPubkeyCompressed(bls_pubkey_compressed_bytes);
+                            BLSPubkey::try_from(bls_pubkey_compressed).ok()
+                        })
```

Comment:
@wen-coding - decided to inline the decompression here for now. I think this is what you suggested originally 🤡 

Can revisit pushing this into `solana-bls-signatures` or something else when we add more consumers

Reply:
Yeah, we keep changing APIs in solana-bls-signatures as we implement BLS sigverify. So we can revisit this later.